cmd/ctr/commands/content/fetch.go

```diff
@@ -101,12 +101,20 @@ func Fetch(ref string, cliContext *cli.Context) (containerd.Image, error) {
 
 	log.G(pctx).WithField("image", ref).Debug("fetching")
 	labels := commands.LabelArgs(cliContext.StringSlice("label"))
-	img, err := client.Pull(pctx, ref,
+	opts := []containerd.RemoteOpt{
 		containerd.WithPullLabels(labels),
 		containerd.WithResolver(resolver),
 		containerd.WithImageHandler(h),
 		containerd.WithSchema1Conversion,
-	)
+	}
+
+	if !cliContext.IsSet("all-platforms") {
```

Comment:
I think you want to get the value for this because checking if a bool is set is not good enough.  You can do things like `--tty=false` where it is set, but value is false